### PR TITLE
Update target and compile sdk version to v29

### PR DESCRIPTION
**Pull request by malinajirka**
This PR updates compile and target sdk version to v29. I checked official android docs, but I haven't found any breaking change which would affect our app - [privacy changes](https://developer.android.com/about/versions/10/privacy/changes) and [behavior changes](https://developer.android.com/about/versions/10/behavior-changes-all). 

- Updates some parts of our code since the nullable/nonnullable annotations were added to some of the java classes in the SDK.
- Suppresses "Field requires API level 29: in MediaStore.MediaColumns" warning. The fields we use were just moved to a super class in the SDK - the constants remain the same.
- Updates gutenberg-mobile reference - corresponding PRs - [gutenberg-mobile](https://github.com/wordpress-mobile/gutenberg-mobile/pull/2646), [gutenberg ](https://github.com/WordPress/gutenberg/pull/25488)

Not ready for merge - related prs in gutenberg repos need to be merged first and the ref needs to be updated

I think the lint is still failing since the changes in gutenberg repo are in a fork - when I targeted the submodule to the fork locally the lint passed without issues.

To test:
Test the app - anything you can think of. Ideally spent at least 10 minutes testing various flows.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by mchowning** [comment]
> I think the lint is still failing since the changes in gutenberg repo are in a fork - when I targeted the submodule to the fork locally the lint passed without issues.

The lint error looks legitimate to me:

> /home/circleci/project/libs/gutenberg-mobile/gutenberg/packages/react-native-aztec/android/src/main/java/org/wordpress/mobile/ReactNativeAztec/ReactAztecManager.java:374: Error: Field requires API level 29 (current min is 21): android.graphics.text.LineBreaker#JUSTIFICATION_MODE_INTER_WORD [InlinedApi]
                view.setJustificationMode(LineBreaker.JUSTIFICATION_MODE_INTER_WORD);
                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> /home/circleci/project/libs/gutenberg-mobile/gutenberg/packages/react-native-aztec/android/src/main/java/org/wordpress/mobile/ReactNativeAztec/ReactAztecManager.java:379: Error: Field requires API level 29 (current min is 21): android.graphics.text.LineBreaker#JUSTIFICATION_MODE_NONE [InlinedApi]
                view.setJustificationMode(LineBreaker.JUSTIFICATION_MODE_NONE);

I left a [comment on the gutenberg PR](https://github.com/WordPress/gutenberg/pull/25488/files#r492380503) about this.

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/utils/StatsIntentUtils.kt
Comment (mchowning) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/utils/StatsIntentUtils.kt:33]: Is this `@Suppress` needed?

Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/utils/StatsIntentUtils.kt:33]: Android studio complains when it's not there - I believe it's a false positive. 
Screenshot 2020-09-23 at 11 37 40

Comment (mchowning) [reply WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/utils/StatsIntentUtils.kt:33]: Interestingly Android Studio doesn't seem to give me any warning if I remove the `@Suppress` line like you did in the screenshot. 🤔

Certainly makes sense to include the `@Suppress` since you're seeing the warning though. 👍

**Review by mchowning — APPROVED** [review]
Tested a number of different flows and everything looks good to me.

⚠️ There is currently a crash when loading the block editor that is unrelated to this pr: p1600864334005800-slack-C6UJ0KRKQ. I expect that will be [fixed soon](https://github.com/wordpress-mobile/gutenberg-mobile/pull/2659) and we just want to retest the block editor on this PR after the gutenberg-mobile submodule ref is updated to make sure we bring in the fix and not just the breakage (basically just make sure the block editor doesn't crash on load).

**Comment by malinajirka** [comment]
> I expect that will be fixed soon and we just want to retest the block editor on this PR after the gutenberg-mobile submodule ref is updated to make sure we bring in the fix and not just the breakage (basically just make sure the block editor doesn't crash on load).

The mentioned PR has been closed (not merged). A fix is included in https://github.com/wordpress-mobile/gutenberg-mobile/pull/2662. We'll need to wait until this new PR gets merged.

**Comment by malinajirka** [comment]
@mchowning I've updated the gutenberg-mobile submodule and merged develop into this branch. I've smoke tested the editor + some other parts of the app and all seems to work as expected. Can I just merge the PR when CI finishes or do you want to check the changes before the merge?

Thank you so much for all your help!!!!! 🙇  🙇  🙇

**Pull request merged**
